lib/python/Pyxsim/__init__.py

```diff
@@ -10,16 +10,49 @@
 import os
 import re
 import sys
+from pathlib import Path
+from typing import Sequence
+import subprocess
 
 from Pyxsim.xmostest_subprocess import call_get_output
 from . import pyxsim
 
 clean_only = False
 
+def cmake_build(xe_path, bin_child:str, env={}, board="XCORE-AI-EXPLORER"):
```

Comment:
needs do_clean functionality

Reply:
Not sure I'm happy about referencing specific HW in here.

Reply:
Should build() take a cmake param rather than a separate function?

Reply:
> Not sure I'm happy about referencing specific HW in here.

Definitely agreed, have removed this as there's better ways of doing it anyway.

> Should build() take a cmake param rather than a separate function?

I'm torn - on the one hand that would be cleaner, on the other I'm not certain that both should have e.g. the same outputs, although again this is mainly from not knowing how else this repo is used in the wild and not wanting to disrupt existing testing schema - thoughts and opinions very welcome here

Reply:
> 

I think it is possible to pass cmake param to _build(). it does the same thing as cmake_build().
for example: you can modify the _build() to
```
def _build(xe_path, build_config=None, env={}, cmake=False do_clean=False, clean_only=False, build_options=[]):
    # Work out the Makefile path
    path = None
    if cmake:
        # Work out your cmakelists_path
        path = cmakelists_path
    else:
        m = re.match("(.*)/bin/(.*)", xe_path)
        if m:
            path = m.groups(0)[0]
            binpath = m.groups(0)[1]
            m = re.match("(.*)/(.*)", binpath)
            if m:
                build_config = m.groups(0)[0]
```
above change won't affect the current test schema.
the only thing I concern is stdout=subprocess.DEVNULL in subprocess.run(make_cmd, cwd=cmakelists_path, env=my_env, stdout=subprocess.DEVNULL). _build() do have a output or error returned by call_get_output and do analyzing and return build success or not. if you dont' need it , simply use if else to skip the build success check.